### ISSUE TITLE
BC PR B: cross-entropy loss + BcTrainer epoch loop + smoke test

### ISSUE THREAD

**Issue by rjwalters** [rjwalters/thrust#167]
**Part of #161** (BC epic). **Depends on #164** (Demonstrations dataset + BcConfig). Labeled `loom:blocked` until #164 merges.

## Background
With the `Demonstrations` dataset and `BcConfig` in place (#164), this PR adds the supervised cross-entropy loss and the `BcTrainer` epoch loop — the core of Behavioral Cloning. Per epic decision 3 (#161), BC reuses `MlpBurnPolicy` (training only its action/logits head via cross-entropy; the value head is ignored), `BurnOptimizer`, and PPO's public `generate_minibatch_indices` for the seeded shuffle. It DIVERGES from every RL trainer in loop shape: a supervised epoch loop over a fixed dataset, no environment interaction, no advantages/returns/entropy. This is PR B of the BC epic.

### 1. `bc/loss.rs` — supervised cross-entropy
```rust
/// -mean(log_softmax(logits).gather(actions)) — NLL of the expert action.
pub fn compute_bc_loss<B: Backend>(logits: Tensor<B, 2>, actions: Tensor<B, 1, Int>) -> Tensor<B, 1>;
```
Discrete-only (matches `MlpBurnPolicy`). Re-use `scalar_f64` (re-exported via `a2c::loss` / `ppo::loss`) for host-side stat extraction.

### 2. `bc/trainer.rs` — `BcTrainer` + `BcEpochStats`
```rust
pub struct BcEpochStats { pub loss: f64, pub accuracy: f64 }

pub struct BcTrainer<B, P, O> { /* config, Option<P>, optimizer, rng, total_steps, total_epochs */ }

impl<B, P, O> BcTrainer<B, P, O> {
    pub fn new(config: BcConfig, policy: P, optimizer: BurnOptimizer<B, P, O>) -> Result<Self>;
    pub fn config(&self) -> &BcConfig;
    pub fn policy(&self) -> &P;
    pub fn total_steps(&self) -> usize;   // gradient updates (one per minibatch)
    pub fn total_epochs(&self) -> usize;
    /// Shuffle (seeded) indices, iterate minibatches of `batch_size`, compute
    /// cross-entropy, backprop, step the optimizer once per minibatch. Returns
    /// mean loss + mean action-match accuracy over the epoch.
    pub fn train_epoch<F>(&mut self, demos: &Demonstrations, forward_fn: F) -> Result<BcEpochStats>
    where F: FnMut(&P, Tensor<B, 2>) -> Tensor<B, 2>; // (policy, obs) -> logits
}
```
- Use the same `Option<P>` move-through-`step` ownership pattern as `A2cTrainer`/`PPOTrainerBurn` (Burn's `Optimizer::step` consumes the module by value).
- Shuffle via PPO's `generate_minibatch_indices` (or an equivalent seeded shuffle driven by the trainer's `StdRng`), so two runs with the same `seed` produce identical minibatch order.
- Accuracy = fraction of minibatch examples whose argmax-logit action matches the expert label, averaged over the epoch.
- `forward_fn` lets the caller pick how logits are produced (e.g. `|p, o| p.forward(o).0` to drop the value head).

### 3. Always-on fast smoke test (`#[cfg(test)]` in `bc/trainer.rs`)
On `Autodiff<NdArray<f32>>`: build a tiny synthetic `Demonstrations` (a handful of `(obs, action)` rows), construct a `BcTrainer`, run ONE `train_epoch`, assert (a) `total_steps() >= 1`, (b) `stats.loss.is_finite()`, (c) `stats.accuracy.is_finite()` and in `[0, 1]`. Fast, deterministic, no feature gate beyond `training`.

### 4. Module wiring
Add `pub mod loss;` and `pub mod trainer;` to `src/train/bc/mod.rs` with re-exports (`BcTrainer`, `BcEpochStats`, `compute_bc_loss`); extend the `src/train/mod.rs` re-exports.

## Acceptance criteria
- `compute_bc_loss` returns a finite scalar; a unit test confirms loss decreases when logits are pushed toward the expert labels (sanity, mirroring `a2c/loss.rs` loss-direction tests).
- `BcTrainer::train_epoch` performs `ceil(len / batch_size)` gradient steps per epoch and returns finite `loss` + `accuracy in [0, 1]`.
- Fast smoke test passes on every `cargo test --features training` run.
- Two `BcTrainer`s built with the same `BcConfig::seed` and same demos produce identical first-epoch stats (reproducibility).
- Zero edits to existing trainers (PPO/DQN/SAC/A2C).

## Affected files
- NEW `src/train/bc/loss.rs`, `src/train/bc/trainer.rs`
- EDIT `src/train/bc/mod.rs`, `src/train/mod.rs` (re-exports)

## References
- Trainer pattern (ownership, `Option<P>`, optimizer step, seeded RNG, stats struct): `src/train/a2c/trainer.rs`.
- Loss pattern + `scalar_f64` re-export: `src/train/a2c/loss.rs`, `src/train/ppo/loss.rs`.
- Logits/log_softmax/gather: `src/policy/mlp.rs` (`forward`, `evaluate_actions`).
- Seeded minibatch shuffle: `generate_minibatch_indices` (re-exported from `src/train/ppo`).
